<!-- jplaced
 2022, Universidad de Zaragoza -->

<!-- Launch file for the random frontier exploration -->
<launch>

    <param name="namespace_init_count"  value="1"/>
    <param name="use_sim_time"          value="true" />

    <arg name="map_topic_name"      value="/gridmapper/rectified_map"/>
    <arg name="global_frame"        value="map"/>
    <arg name="robot_namespace"     value="robot_1"/>
    <arg name="robot_number"        value="1"/>
    <arg name="camInfo_topic_name"  value="/$(arg robot_namespace)/camera/rgb/camera_info"/>
    <arg name="gpu"                 value="false" />
    <arg name="camera_type"         value="rgbd" /> <!-- 'rgbd' or 'stereo' -->

    <include file="$(find frontier_detector)/launch/frontiers.launch">
        <arg name="map_topic_name"  value="$(arg map_topic_name)"/>
        <arg name="global_frame"    value="$(arg global_frame)"/>
        <arg name="robot_namespace" value="$(arg robot_namespace)"/>
        <arg name="gpu"             value="$(arg gpu)"/>
        <arg name="camera_type"     value="$(arg camera_type)"/>
    </include>

    <node pkg="decision_maker" type="GraphDrawer.py" name="G_publisher" output="screen">
        <param name="camera_type"   value="$(arg camera_type)"/>
    </node>

    <node pkg="decision_maker" type="Controller.py" name="decision_maker" output="screen">
        <param name="map_topic"              value="$(arg map_topic_name)"/>
        <param name="cameraInfo_topic"       value="$(arg camInfo_topic_name)"/>
        <param name="global_frame"           value="$(arg global_frame)"/>
        <param name="frontiers_topic"        value="/frontier_detectors/filter/filtered_points"/>
        <param name="n_robots"               value="$(arg robot_number)"/>
        <param name="delay_after_assignment" value="0.5"/>
        <param name="rate"                   value="10"/>
        <param name="type"                   value="random"/>
        <param name="show_debug_path"        value="true"/>
        <param name="exploring_time"         value="9000"/>
        <param name="enable_gpu_comp"        value="$(arg gpu)"/>
        <param name="hallucinated_plan_th"   value="60"/>
        <param name="camera_type"            value="$(arg camera_type)"/>
    </node>

</launch>
